application/src/main/java/org/thingsboard/server/service/install/SqlDatabaseUpgradeService.java

```diff
@@ -752,6 +752,11 @@ public void upgradeDatabase(String fromVersion) throws Exception {
                             conn.createStatement().execute("ALTER TABLE rule_node ADD COLUMN IF NOT EXISTS configuration_version int DEFAULT 0;");
                         } catch (Exception e) {
                         }
+                        try {
+                            conn.createStatement().execute("UPDATE rule_node SET configuration = jsonb_set(configuration::jsonb, '{updateAttributesOnlyOnValueChange}', '\"false\"'::jsonb), " +
```

Comment:
Please simplify using this SO answer: https://stackoverflow.com/questions/13615760/add-element-to-json-object-in-postgres.

Reply:
Also don't forget to cast the jsonb back to text.

Reply:
> Also don't forget to cast the jsonb back to text.

I have tested this locally without casting the jsonb back to text, and it works as expected. PostgreSQL appears to perform the necessary conversion behind the scenes to update the VARCHAR column correctly.

Reply:
Anyway, I will add explicit type cast back to varchar to avoid unexpected behaviors in more complex scenarios that might use this upgrade entry as an example.